### DISCUSSION TITLE
crypto_secretbox algorithm stability over releases

### DISCUSSION THREAD

**Discussion by MathieuBordere** [jedisct1/libsodium#1394] [General]
I'm planning on using libsodium to encrypt data on disk for a possible long period of time. I assume the algorithm behind `crypto_secretbox` can change depending on the release? So it's better to use the underlying algorithm directly if I plan on updating libsodium regularly?

Thanks,
Mathieu

**Comment by jedisct1** [comment]
https://libsodium.gitbook.io/doc/quickstart#should-i-call-crypto_generichash_blake2b-or-just-crypto_generichash
